scrape/manager.go

```diff
@@ -205,10 +205,19 @@ func (m *Manager) reload() {
 			sp.Sync(groups)
 			wg.Done()
 		}(m.scrapePools[setName], groups)
-
 	}
 	m.mtxScrape.Unlock()
 	wg.Wait()
+
+	activeTargets := make(map[uint64]*Target)
+	for _, scrapePool := range m.scrapePools {
+		for _, target := range scrapePool.activeTargets {
+			if t, ok := activeTargets[target.hash()]; ok {
+				level.Warn(m.logger).Log("msg", "Found targets with same labels after relabelling", "target", t, "target", target)
```

Comment:
The issue here is that the identifiers for the targets (`t` and `target`) are a 64-bit integer and a URL, the former being not very helpful, and both not matching.

I'm partial to just including the URL of the target, and because both will be the same we only need one, but I'll turn to @roidelapluie for a second opinion.

Reply:
Actually, my comment is wrong. Both of these come out to be the URL of the target, and since that's the same, we only need one.